lib/utilities/retrieveData.js

```diff
@@ -60,7 +60,17 @@ export default function retrieveData(locations, retrieveDataFn, response, log) {
                 log.error('error piping data from source');
                 return _next(err);
             });
-            return readable.pipe(response, { end: false });
+            readable.on('data', chunk => {
+                const write = response.write(chunk);
+                if (!write) {
+                    readable.pause();
+                    response.once('drain', () => {
+                        response.write(chunk);
```

Comment:
won't this write the same chunk again?  The docs say -- "false if the stream wishes for the calling code to wait for the 'drain' event to be emitted before continuing to write **additional** data"

Reply:
Instead of `response.write` you may want to call `readable.resume()` here

Reply:
My bad, you call it just below, so removing `response.write` should be fine.

Reply:
I was under the assumption that I had to write the same chunk again...reading the docs looks like it does accept the chunk and we have to pause for additional data. Thanks for catching that!